serveradmin/dataset/queryset.py

```diff
@@ -88,15 +88,9 @@ def _get_query_builder(self):
                     if s.ip_addr_type in desired_ip_addr_types(filt)
                 }
 
+            # We can just deal with the servertype filters ourself.
             if attribute.pk == 'servertype':
-                if isinstance(filt, str):
-                    servertypes = {
-                        s for s in servertypes if s.pk == filt
-                    }
-                else:
-                    servertypes = {
-                        s for s in servertypes if filt.matches(s.pk)
-                    }
+                servertypes = {s for s in servertypes if filt.matches(s.pk)}
```

Comment:
That is also possible because of the commit before, right?

Reply:
Yes.  It is the same commit actually.